java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java

```diff
@@ -3890,6 +3882,33 @@ && isOfKindAndType(e.getEnclosingElement().asType(), e, kinds, baseType, scope,
                 }
                 return false;
             }
+
+            private boolean checkMethodAccepted(String sn, Element e,
+                    TypeMirror t) {
+                //4 checks to include property accessors
+                String initialPrefix = env.getPrefix();
+                //TODO: maybe cache these?
+                String[] prefixes = {
+                    initialPrefix,
+                    "is" + initialPrefix,
+                    "get" + initialPrefix,
+                    "set" + initialPrefix
+                };
+                for (String prefix : prefixes) {
+                    if (checkMethodAccepted(sn, prefix, e, t)){
+                        return true;
+                    }
+                }
+                return false;
```

Comment:
alternative which might be more readable:
```java
return checkMethodAccepted(sn, initialPrefix, e, t)
    || checkMethodAccepted(sn, "is" + initialPrefix, e, t)
    || checkMethodAccepted(sn, "get" + initialPrefix, e, t)
    || checkMethodAccepted(sn, "set" + initialPrefix, e, t)
```

I am a little worried about performance here. We recently made completion in cases with large amount of fields faster (#4142).
We will have to check that this is not affecting performance negatively.

Here a file for manual tests: https://github.com/apache/netbeans/issues/4140#issuecomment-1134414691

Reply:
I see... I could possibly take a look at it later... But I don't know if I will have the time. That logic is only invoked for filtering methods though, not fields, but I see your point.